### ISSUE TITLE
Retrieving meshes from the plotter object

### ISSUE THREAD

**Issue by germa89** [pyvista/pyvista#4769]
### Describe the bug, what's wrong, and what you expected.

I'm not sure if this is a bug or it is the intended behaviour and I'm using pyvista wrong.

But if I add several meshes to a plotter, when I use `pl.mesh` I only retrieve the latest mesh. IMO it should retrieve either a mesh which combine all all the previous meshes, or a list of those meshes.

Maybe am I using it wrong?

### Steps to reproduce the bug.

```python

import pyvista

sphere = pyvista.Sphere(center=(1, 0, 0))
box = pyvista.Box()

pl = pyvista.Plotter()
pl.add_mesh(sphere, scalars=range(sphere.n_points))
pl.add_mesh(box, scalars=range(box.n_cells))

assert box == pl.mesh
```

### System Information

```shell
--------------------------------------------------------------------------------
  Date: Mon Aug 14 17:58:38 2023 CEST

                OS : Darwin
            CPU(s) : 8
           Machine : arm64
      Architecture : 64bit
               RAM : 16.0 GiB
       Environment : IPython
       File system : apfs
        GPU Vendor : Apple
      GPU Renderer : Apple M2
       GPU Version : 4.1 Metal - 83.1
  MathText Support : False

  Python 3.11.3 (main, Apr  7 2023, 20:13:31) [Clang 14.0.0
  (clang-1400.0.29.202)]

           pyvista : 0.41.1
               vtk : 9.2.6
             numpy : 1.25.1
        matplotlib : 3.7.2
            scooby : 0.7.2
             pooch : v1.7.0
           imageio : 2.31.1
           IPython : 8.14.0
        ipywidgets : 8.0.7
             scipy : 1.11.1
              tqdm : 4.65.0
            meshio : 5.3.4
        jupyterlab : 4.0.2
         pythreejs : 2.4.2
             trame : 3.1.0
      trame_client : 2.10.0
      trame_server : 2.11.7
         trame_vtk : 2.5.8
      nest_asyncio : 1.5.6
--------------------------------------------------------------------------------
```

**Comment by akaszynski** [comment]
Intended behavior is for `Plotter.mesh` to just be the last mesh. It's not the best API, and we might deprecate in the future.

If you want to access the meshes (datasets), access the dataset attribute of each actor's mapper:
```py
datasets = []
for actor in pl.actors.values():
    if hasattr(actor, 'mapper'):
        datasets.append(actor.mapper.dataset)

datasets = [actor.mapper.dataset for actor in pl.actors.values() if hasattr(actor, 'mapper')]
```

**Comment by germa89** [comment]
Thank you a lot @akaszynski 

I would propose then to have `pl.meshes` to access the meshes. I could do the implementation based on your snippet, if you want.

**Comment by akaszynski** [comment]
Works for me!

**Comment by germa89** [comment]
For keeping track record.

I believe the following methods should have the following usage:
* `Plotter.meshes` -> Return a list of mesh object. Implemented in #4779
* `Plotter.mesh` -> Return a mesh object which is the result of merging all the meshes added to the plotter.
```py
def mesh(self):
    return merge_meshes(self.meshes)  # `merge_meshes` should be created.
```
* `Plotter.last_mesh` -> Return last mesh. Mimic current `Plotter.mesh` method.
```py
def last_mesh(self):
    return self.meshes[-1]
```
